File: Kata/HumanTimeFormat.cs
using System.Linq;

namespace Kata
{
    public class HumanTimeFormat
    {
        private const int Minute = 60;
        private const int Hour = Minute * 60;
        private const int Day = Hour * 24;
        private const int Year = Day * 365;

        public static string formatDuration(int seconds)
        {
            if (seconds == 0)
                return "now";

            string plural(int x)
            {
                return x > 1 ? "s" : null;
            }

            string output(int x, string s)
            {
                return x > 0 ? $"{x} {s}{plural(x)}" : null;
            }

            var y = seconds / Year;
            seconds -= y * Year;
            var d = seconds / Day;
            seconds -= d * Day;
            var h = seconds / Hour;
            seconds -= h * Hour;
            var m = seconds / Minute;
            seconds -= m * Minute;

            var s1 = string.Join
            (
                ", ", 
                new[] 
                {
                    output(y, "year"), 
                    output(d, "day"), 
                    output(h, "hour"), 
                    output(m,"minute"),
                    output(seconds, "second")
                }.Where(s => s != null)
            );

            var cIndex = s1.LastIndexOf(',');
            if (cIndex != -1)
                s1 = s1.Substring(0, cIndex) + " and" + s1.Substring(cIndex + 1);

            return s1;
        }
    }
}

File: PythonKata/HumanTimeFormat.py
def format_duration(seconds):
    if seconds == 0:
        return "now"

    Minute : int = 60;
    Hour : int = Minute * 60;
    Day : int = Hour * 24;
    Year : int = Day * 365;

    y : int = int(seconds / Year);
    seconds -= y * Year;
    d : int = int(seconds / Day);
    seconds -= d * Day;
    h : int = int(seconds / Hour);
    seconds -= h * Hour;
    m : int = int(seconds / Minute);
    seconds -= m * Minute;

    outStringList = list(filter(lambda s: s != None, [output(y, "year"), output(d, "day"), output(h, "hour"), output(m, "minute"), output(seconds, "second")]))
    outString = ""
    l = len(outStringList)

    for i in range(0, l):
        outString += ", " if i > 0 and i < l - 1 else " and " if i > 0 and i == l - 1 else ""
        outString += outStringList[i]

    return outString

def plural(x : int):
    return "s" if x > 1 else ""

def output(x : int, s):
    return str(x) + " " + s + plural(x) if x > 0 else None

